x-pack/plugins/security_solution/server/lib/detection_engine/routes/__mocks__/rule_status_service.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SavedObjectsClientContract } from '../../../../../../../../src/core/server';
+import { RuleStatusSavedObjectsClient } from '../../signals/rule_status_saved_objects_client';
+import { RuleStatusService } from '../../signals/rule_status_service';
+
+export type RuleStatusServiceMock = jest.Mocked<RuleStatusService>;
+
+export const ruleStatusServiceFactoryMock = async ({
+  alertId,
+  ruleStatusClient,
+}: {
+  alertId: string;
+  ruleStatusClient: RuleStatusSavedObjectsClient;
+}): Promise<RuleStatusServiceMock> => {
+  return {
+    goingToRun: jest.fn(),
+
+    success: jest.fn(),
+
+    error: jest.fn(),
+  };
+};
+
+export type RuleStatusSavedObjectsClientMock = jest.Mocked<RuleStatusSavedObjectsClient>;
+
+export const ruleStatusSavedObjectsClientFactory = (
+  savedObjectsClient: SavedObjectsClientContract
+): RuleStatusSavedObjectsClientMock => ({
+  find: jest.fn(),
+  create: jest.fn(),
+  update: jest.fn(),
+  delete: jest.fn(),
+});
```

Comment:
I would move this file to be `rule_status_service.mocks.ts` and reduce usage of the `__mocks__` folder as that is what the rest of Kibana is doing these days.